components/infobox/commons/infobox_person.lua

```diff
@@ -0,0 +1,280 @@
+		else
+			infobox:categories('Active ' .. role.category .. 's')
+		end
+		if not args.id then
+			infobox:categories('InfoboxIncomplete.')
+		end
+		if not args.image then
+			infobox:categories(role.category .. 's with no profile picture')
+		end
+		if not birthDisplay then
+			infobox:categories(role.category .. 's with unknown birth date')
+		end
+
+		links = self:_getLinksLPDB(links)
+		local lpdbData = {
+			id = args.id or mw.title.getCurrentTitle().prefixedText,
+			alternateid = args.ids,
+			name = args.romanized_name or args.name,
+			localizedname = args.name,
+			nationality = args.country or args.nationality,
+			nationality2 = args.country2 or args.nationality2,
+			nationality3 = args.country3 or args.nationality3,
+			birthdate = birthday,
+			deathdate = deathday,
+			image = args.image,
+			region = args.region,
+			team = args.teamlink or args.team,
+			status = status.store,
+			type = role.store,
+			earnings = earnings,
+			links = links,
+			extradata = {},
+		}
+		lpdbData = self:adjustLPDB(lpdbData, args, role, status)
+		lpdbData.extradata = mw.ext.LiquipediaDB.lpdb_create_json(lpdbData.extradata)
+		lpdbData.links = mw.ext.LiquipediaDB.lpdb_create_json(lpdbData.links)
+		local storageType = self:getStorageType(args, role, status)
+
+		mw.ext.LiquipediaDB.lpdb_player(storageType .. self.name, lpdbData)
+	end
+
+	return infobox:build()
+end
+
+--- Allows for overriding this functionality
+function Player:getStorageType(args, role, status)
+	return 'player'
+end
+
+--- Allows for overriding this functionality
+function Player:getInformationType(args)
+	return args.informationType or 'Player'
+end
+
+--- Allows for overriding this functionality
+function Player:adjustLPDB(lpdbData, args, role, status)
+	return lpdbData
+end
+
+--- Allows for overriding this functionality
+function Player:getRole(args)
+	return { display = args.role, store = args.role, category = args.role or 'Player'}
+end
+
+--- Allows for overriding this functionality
+function Player:getStatus(args)
+	return { display = args.status, store = args.status }
+end
+
+--- Allows for overriding this functionality
+function Player:getHistory(infobox, args)
+	return args.history
+end
+
+--- Allows for overriding this functionality
+--- Decides if we store in LPDB and Vars or not
+function Player:shouldStoreData(args)
+	return Namespace.isMain()
+end
+
+--- Allows for overriding this functionality
+--- e.g. to add faction icons to the display for SC2, SC, WC
+function Player:nameDisplay(args)
+	local team = args.teamlink or args.team
+	local icon = mw.ext.TeamTemplate.teamexists(team)
+		and mw.ext.TeamTemplate.teamicon(team) or ''
+	local name = args.id or mw.title.getCurrentTitle().text
+
+	return icon .. '&nbsp;' .. name
+end
+
+--- Allows for overriding this functionality
+function Player:getAchievements(infobox, args)
+	return args.achievements
+end
+
+--- Allows for overriding this functionality
+function Player:calculateEarnings(args)
+	return 0
```

Comment:
Let's make this an error, like in Infobox/Team

Reply:
the idea to not have an error here is that we want to use this infobox for non players too
for e.g. casters, mapmakers, ... earnings is not needed
the return of 0 will make it so that it will not be displayed

alternatively we would need to override this function in any of those modules
and the /Player one needs a custom function anyways

Reply:
Alright. In that case 
```suggestion
	return nil
```

Reply:
that would cause an error at line 40 (adding the `$` and formatting the value if the value is not 0)
so either let it return 0 or on line 39 we need to change the `else` to `elseif earnings then`

Reply:
Well we already detect the 0, no?

```lua
	if earnings == 0 then
		earnings = nil
	else
```

so just remove this and make it the following if:

```lua
if earnings ~= nil and earnings ~= 0
```